src/Windows/Avalonia.Win32/OleDropTarget.cs

```diff
@@ -1,18 +1,19 @@
 ﻿using Avalonia.Input;
 using Avalonia.Input.Raw;
+using Avalonia.MicroCom;
 using Avalonia.Platform;
 using Avalonia.Win32.Interop;
-using IDataObject = Avalonia.Input.IDataObject;
+using DropEffect = Avalonia.Win32.Win32Com.DropEffect;
 
 namespace Avalonia.Win32
 {
-    internal class OleDropTarget : IDropTarget
+    internal class OleDropTarget : CallbackBase, Win32Com.IDropTarget
     {
         private readonly IInputRoot _target;
         private readonly ITopLevelImpl _tl;
         private readonly IDragDropDevice _dragDevice;
         
-        private IDataObject _currentDrag = null;
+        private IDisposableDataObject _currentDrag = null;
```

Comment:
I added IDisposableDataObject so both DataObject and OleDataObject can be disposed on drag-leave or drop events. But I think I was wrong here, and only OleDataObject should be disposed, because it's not reused and keeps native reference. While DataObject is actually reused and can be disposed twice (which is no-op in current implementation, but still).

Reply:
I will probably do checks to dispose only DataObject in this class.